bundle/tests/bundle/pipeline_glob_paths_test.go

```diff
@@ -32,13 +26,30 @@ func TestExpandPipelineGlobPathsWithNonExistent(t *testing.T) {
 	}, nil)
 	b.SetWorkpaceClient(m.WorkspaceClient)
 
-	err = bundle.Apply(ctx, b, phases.Initialize())
-	require.Error(t, err)
-	require.ErrorContains(t, err, "notebook ./non-existent not found")
-
+	ctx := context.Background()
+	err := bundle.Apply(ctx, b, phases.Initialize())
+	require.NoError(t, err)
 	require.Equal(
 		t,
-		b.Config.Resources.Pipelines["nyc_taxi_pipeline"].Libraries[0].Notebook.Path,
 		"/Users/user@domain.com/.bundle/pipeline_glob_paths/default/files/dlt/nyc_taxi_loader",
+		b.Config.Resources.Pipelines["nyc_taxi_pipeline"].Libraries[0].Notebook.Path,
 	)
 }
+
+func TestExpandPipelineGlobPathsWithNonExistent(t *testing.T) {
+	b := loadTarget(t, "./pipeline_glob_paths", "error")
+
+	// Configure mock workspace client
+	m := mocks.NewMockWorkspaceClient(t)
+	m.WorkspaceClient.Config = &config.Config{
+		Host: "https://mock.databricks.workspace.com",
+	}
+	m.GetMockCurrentUserAPI().EXPECT().Me(mock.Anything).Return(&iam.User{
+		UserName: "user@domain.com",
+	}, nil)
+	b.SetWorkpaceClient(m.WorkspaceClient)
+
+	ctx := context.Background()
+	err := bundle.Apply(ctx, b, phases.Initialize())
+	require.ErrorContains(t, err, "notebook ./non-existent not found")
+}
```

Comment:
These changes were needed because this change no longer has an observable side effect on the configuration if it is partially OK. If there is an error, none of the path translations are persisted in the config.